tests/testautotls.nim

```diff
@@ -0,0 +1,140 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.push raises: [].}
+
+import std/[strformat, net]
+import chronos
+import chronos/apps/http/httpclient
+import
+  ../libp2p/[
+    stream/connection,
+    transports/tcptransport,
+    upgrademngrs/upgrade,
+    multiaddress,
+    switch,
+    builders,
+    autotls/autotls,
+    autotls/acme,
+    autotls/utils,
+    nameresolving/dnsresolver,
+    wire,
+  ]
+
+import ./helpers
+
+suite "AutoTLS":
+  asyncTest "test ACME":
+    let acc = await ACMEAccount.new(
+      KeyPair.random(PKScheme.RSA, newRng()[]).get(),
+      acmeServerURL = LetsEncryptURLStaging,
+    )
+    defer:
+      checkTrackers()
+
+    await acc.register()
+    # account was registered (kid set)
+    check acc.kid.isSome
+
+    # challenge requested
+    let (dns01Challenge, finalizeURL, orderURL) =
+      await acc.requestChallenge(@["some.dummy.domain.com"])
+    check dns01Challenge.isNil == false
+    check finalizeURL.len > 0
+    check orderURL.len > 0
+    await noCancel(acc.session.closeWait())
+
+  asyncTest "test AutoTLSManager":
+    let switch = SwitchBuilder
+      .new()
+      .withRng(newRng())
+      .withAddress(MultiAddress.init("/ip4/0.0.0.0/tcp/0").tryGet())
+      .withTcpTransport()
+      .withAutoTLSManager(acmeServerURL = LetsEncryptURLStaging)
+      .withYamux()
+      .withNoise()
+      .build()
+
+    defer:
+      checkTrackers()
```

Comment:
convention in code is currently is to have `checkTrackers` in `teardown`. it would be nice to follow the same convention.

Reply:
Did you not want me to remove `setup` and `teardown` in https://github.com/vacp2p/nim-libp2p/pull/1358#discussion_r2108735434?